e3sm_diags/driver/utils/regrid.py

```diff
@@ -456,6 +447,42 @@ def _drop_unused_ilev_axis(ds: xr.Dataset) -> xr.Dataset:
     return ds_new
 
 
+def _align_bounds_values(
+    ds_a_new: xr.Dataset, ds_b_new: xr.Dataset
+) -> Tuple[xr.Dataset, xr.Dataset]:
+    """Align bounds values between two datasets if they are not equal.
+
+    This is essential for xESMF's `conservative_normed` regridding, which
+    relies on aligned lat/lon bounds for accuracy. Misaligned bounds can
+    cause significant differences due to cell fraction mismatch, interpolation
+    artifacts, weighting inconsistencies, and/or edge effects.
+
+    Parameters
+    ----------
+    ds_a_new : xr.Dataset
+        The first dataset.
+    ds_b_new : xr.Dataset
+        The second dataset.
+
+    Returns
+    -------
+    Tuple[xr.Dataset, xr.Dataset]
+        A tuple of both datasets with aligned bounds values.
+    """
+    for axis in ["X", "Y"]:
+        bounds_a = ds_a_new.bounds.get_bounds(axis=axis)
+        bounds_b = ds_b_new.bounds.get_bounds(axis=axis)
+
+        if not np.array_equal(bounds_a.values, bounds_b.values):
+            ds_a_new = ds_a_new.drop_vars(bounds_a.name)
+            ds_b_new = ds_b_new.drop_vars(bounds_b.name)
+
+    ds_a_new = ds_a_new.bounds.add_missing_bounds(axes=["X", "Y"])
+    ds_b_new = ds_b_new.bounds.add_missing_bounds(axes=["X", "Y"])
+
```

Comment:
RE: https://github.com/E3SM-Project/e3sm_diags/pull/940#issuecomment-2707530432

This logic will always drop bounds if the model and obs have different grid sizes (and bounds shape).
I need to add update it so that it drops bounds if the bounds are the same shape but with non-equal values. 

I wonder if this has any affect on the RMSE for land averaged T with ERA5. I will isolate this variable and debug.

Reply:
yeah, I think the logic needs to be refined. Though i think it is good to always have xcdat to generate bounds (maybe more reliable?).The ERA5 data is on 0.25 degree resolution, and model is at 1 degree. I'm doing some offline analysis to see the difference between bilinear and conservative_normed. Maybe it is actually expected to see the difference. If it is this case, we may need to decide on if we actually want `conservative` on this field or not. In general use `bilinear` for a state variable such as temperature should be good. I'm going to upload the debugging notebook as well.

Reply:
Updated logic here: https://github.com/E3SM-Project/e3sm_diags/pull/940/commits/24c065741bb1df5244e6c8553d0a665401a6e6ef